src/lib/core/WeaveConfig.h

```diff
@@ -1403,6 +1404,55 @@
 #define WEAVE_CONFIG_DEBUG_CERT_VALIDATION                  1
 #endif // WEAVE_CONFIG_DEBUG_CERT_VALIDATION
 
+/**
+ *  @def WEAVE_CONFIG_DEFAULT_CERT_CURVE_ID
+ *
+ *  @brief
+ *    Default curve to be used to generate Weave operational device certificate.
+ *
+ */
+#ifndef WEAVE_CONFIG_DEFAULT_CERT_CURVE_ID
+#if WEAVE_CONFIG_SUPPORT_ELLIPTIC_CURVE_SECP256R1
+#define WEAVE_CONFIG_DEFAULT_CERT_CURVE_ID                  (nl::Weave::Profiles::Security::kWeaveCurveId_prime256v1)
+#elif WEAVE_CONFIG_SUPPORT_ELLIPTIC_CURVE_SECP224R1
+#define WEAVE_CONFIG_DEFAULT_CERT_CURVE_ID                  (nl::Weave::Profiles::Security::kWeaveCurveId_secp224r1)
+#elif WEAVE_CONFIG_SUPPORT_ELLIPTIC_CURVE_SECP192R1
+#define WEAVE_CONFIG_DEFAULT_CERT_CURVE_ID                  (nl::Weave::Profiles::Security::kWeaveCurveId_prime192v1)
+#elif WEAVE_CONFIG_SUPPORT_ELLIPTIC_CURVE_SECP160R1
+#define WEAVE_CONFIG_DEFAULT_CERT_CURVE_ID                  (nl::Weave::Profiles::Security::kWeaveCurveId_secp160r1)
```

Comment:
Let make the default just P256, and have compilation fail if P256 is not supported.  That way developers have to consciously decide to use a smaller curve.

Reply:
updated to use P256 curve